farcaster/modules/airstack/social/social.ts

```diff
@@ -0,0 +1,274 @@
+import { BigInt, ethereum } from '@graphprotocol/graph-ts';
+import {
+  AirExtraData,
+  AirUser,
+  AirProfile,
+  AirBlock,
+  AirUserRegisteredTransaction,
+} from '../../../generated/schema';
+import { getOrCreateAirAccount, getOrCreateAirBlock, getChainId, updateAirEntityCounter } from '../common/index';
+import { AirProtocolType, AirProtocolActionType, AIR_USER_REGISTERED_TRANSACTION_ENTITY_COUNTER_ID } from './utils';
+
+export namespace social {
+
+  /**
+   * @dev this function tracks a air user and profile registered transaction
+   * @param blockNumber block number
+   * @param blockHash block hash
+   * @param blockTimestamp block timestamp
+   * @param transactionHash transaction hash
+   * @param logOrCallIndex log or call index
+   * @param fromAddress air transaction from which user token was transferred
+   * @param toAddress air transaction to which user token was transferred
+   * @param tokenId tokenId of the profile token that was transferred - ERC721
+   * @param dappUserId dappUserId (eg: farcasterId)
+   * @param profileName profile name
+   * @param profileExtras extra data (eg: farcaster tokenUri)
+   * @param userExtras extra data (eg: farcaster homeUrl and recoveryAddress)
+   */
+  export function trackUserAndProfileRegisteredTransaction(
+    blockNumber: BigInt,
+    blockHash: string,
+    blockTimestamp: BigInt,
```

Comment:
We can have the above 3 fields in one as `block`

Reply:
done